tf_quant_finance/math/pde/fd_solvers.py

```diff
@@ -81,63 +81,76 @@ def solve_backward(start_time,
 
   ### Example. European call option pricing.
   ```python
+  import tensorflow as tf
+  import tf_quant_finance as tff
+  pde = tff.math.pde
+
   num_equations = 2  # Number of PDE
   num_grid_points = 1024  # Number of grid points
-  dtype = np.float64
+  dtype = tf.float64
+
   # Build a log-uniform grid
   s_min = 0.01
   s_max = 300.
-  grid = uniform_grid(minimums=[s_min], maximums=[s_max],
-                      sizes=[num_grid_points],
-                      dtype=dtype)
+  grid = pde.grids.uniform_grid(minimums=[s_min],
+                                maximums=[s_max],
+                                sizes=[num_grid_points],
+                                dtype=dtype)
+
   # Specify volatilities and interest rates for the options
+  strike = tf.constant([[50], [100]], dtype)
   volatility = tf.constant([[0.3], [0.15]], dtype)
   rate = tf.constant([[0.01], [0.03]], dtype)
   expiry = 1.0
-  strike = tf.constant([[50], [100]], dtype)
+
+  # For batching multiple PDEs, we need to stack the grid values
+  # so that final_values[i] is the grid for the ith strike.
+  # We will use `tf.meshgrid` to quickly do this.
+  s = grid[0]
+  strike_mesh, s_mesh = tf.meshgrid(strike, s, indexing='ij')
+  final_value_grid = tf.nn.relu(s_mesh - strike_mesh)
 
   # Define parabolic equation coefficients. In this case the coefficients
   # can be computed exactly but the same functions as below can be used to
   # get approximate values for general case.
+  # We again need to use `tf.meshgrid` to batch the coefficients.
   def second_order_coeff_fn(t, grid):
-    del t
-    x = grid[0]
-    return [[tf.square(volatility) * tf.square(x) / 2]]
+    s = grid[0]
+    vol_mesh, s_mesh = tf.meshgrid(volatility, s, indexing='ij')
```

Comment:
No need for this, the answer is the same as before. Automatic broadcasting applies:
`volatility` has shape `(2, 1)` and `grid[0]` has shape `(1024)` which have a common shape `(2, 1024)`

Reply:
tf.meshgrid is probably coming from examples in pde_solvers.pdf. But Cyril is right that we should prefer using broadcasting.
I have updated pde_solvers.pdf, so Kyle, if you could update this example accordingly, i.e. remove tf.meshgrid, that would be very helpful!

Reply:
Thanks for the update to pdf_solvers.pdf. This definitely simplifies the code a lot.